iblvideo/run.py

```diff
@@ -1,47 +1,62 @@
-from datetime import datetime
+import logging
+
 from oneibl.one import ONE
 from oneibl.patcher import FTPPatcher
-from .choiceworld import dlc
-from .weights import download_weights
-from . import __version__
+from iblvideo.choiceworld import dlc
+from iblvideo.weights import download_weights
+from iblvideo import __version__
+from ibllib.pipes import tasks
+
+_logger = logging.getLogger('ibllib')
+
+
+# re-using the Task class allows to not re-write all the logging, error management
+# and automatic settings of task statuses in the database
+class TaskDLC(tasks.Task):
+    gpu = 1
+    cpu = 4
+    io_charge = 90
+    level = 0
 
+    def _run(self, version=__version__):
+        # Download weights into ONE Cache directory under 'resources/DLC' if not exist
+        path_dlc = download_weights(version=version)
+        files_mp4 = list(self.session_path.joinpath('raw_video_data').glob('*.mp4'))
+        if len(files_mp4) == 0:
+            raise FileNotFoundError('no video file found')
+        # Run dlc on all three videos and upload data, set status to complete
+        out_files = []
+        for cam in range(len(files_mp4)):
+            dlc_result = dlc(files_mp4[cam], path_dlc)
+            out_files.append(dlc_result)
+            _logger.info(dlc_result)
+        [_logger.info(f) for f in out_files]
+        pqts = list(self.session_path.joinpath('alf').glob('*.pqt'))
+        return pqts
 
-def run_session(session_id, version=__version__, one=None, ftp_patcher=None):
```

Comment:
In this case we need to remove the ftp patcher creation form run_queue I think? Or at least it cannot be passed to run_session anymore